### ISSUE TITLE
Dynamic JS pages/components

### ISSUE THREAD

**Issue by duailibe** [facebook/Docusaurus#292]
Hey guys! Loving this is open now, I hope to be to contribute soon!

We've been using just a [simple JS script](https://github.com/prettier/prettier/blob/8bb6e16/website/static/playground.js) on the Prettier docs playground and I've been pushing for a rewrite in React so we could do some cool stuff more easily: 

- creating the options sidebar automatically so a new PR that changes anything could be more easily integrated to the playground and people can play with it in the PR previews;
- persisting the state in localStorage

To do that a build process would be necessary.. we currently just serve from `/static`, so I built a POC with the features above using Preact and https://github.com/developit/preact-in-es3/blob/master/index.js, but it's really cumbersome without JSX and would be nice to do it in React since that's what Docusaurus uses.

We could just build the playground independently (calling the transpiler/bundler and outputing directly to Docusaurus build dir), but would be nice to integrate this somehow so we could reuse the development server and have one script to build them all.

If this use case is something you guys would like to support, I'd love to help!

**Comment by JoelMarcey** [comment]
@duailibe Thanks for writing this up. While not totally related, I believe @hramos had to do some customization to get the React Native web player running (e.g., http://facebook.github.io/react-native/docs/flexbox.html); so he might have some insight on this as well.

Let me talk this over a bit with the team.

cc: @ericnakagawa @hramos

**Comment by azz** [comment]
We'd really like to be able to hydrate the code on the client-side to support some interactions. Is that on the cards?

**Comment by endiliey** [comment]
I believe this is now achieved in v2, should we close this @yangshun ?

**Comment by SimenB** [comment]
There's a v2? 😀 Any changelog somewhere? The one in the root only mentions v1

**Comment by endiliey** [comment]
Not yet released @SimenB. https://docusaurus.io/blog/2018/09/11/Towards-Docusaurus-2

**Comment by yangshun** [comment]
That's true, but that's considering the next release will be out soon. We don't have a launch date for v2 yet. But I guess we can close it for now and see whether there are any downsides of closing them before actually launching.

**Comment by endiliey** [comment]
Kinda agreed on second thought 🤔. Nah maybe it's time for us to work to launch v2 sooner then before it backfires 😃
